packages/core/src/services/shellExecutionService.ts

```diff
@@ -216,8 +217,18 @@ export class ShellExecutionService {
 
             if (stream === 'stdout') {
               stdout += decodedChunk;
+              if (stdout.length > MAX_CHILD_PROCESS_BUFFER_SIZE) {
```

Comment:
we should notify the user&model if there is truncation. also consider making a helper method that safely appends content to the buffer truncating as needed. This logic is fine in general but would have edge case issues if `stdout + decodedChunk` was large enough to exceed the maximum string size we can handle. It would be instead by slightly more performance and robust to first truncate stdout by the amount needed to fit within the buffer size and only then add decodedChunk. For full robustness would also need to handle if decodedChunk is larger than MAX_CHILD_PROCESS_BUFFER_SIZE